codetransformer/core.py

```diff
@@ -190,4 +211,26 @@ def code(self):
         try:
             return self._code_stack[-1]
         except IndexError:
-            raise NoCodeContext()
+            raise NoContext()
+
+    @property
+    def startcode(self):
+        """The startcode we are currently in.
+        """
+        try:
+            return self._startcode_stack[-1]
+        except IndexError:
+            raise NoContext()
+
+    def begin(self, startcode):
+        """Begin a new startcode.
+
+        Parameters
+        ----------
+        startcode : any
+            The startcode to begin.
+        """
+        try:
+            self._startcode_stack[-1] = startcode
```

Comment:
It's a little bit confusing that we're writing to the last value of a variable called `stack`.